index.js

```diff
@@ -125,7 +125,10 @@ Discovery.prototype.destroy = function (cb) {
     self.tracker.removeListener('peer', self._onTrackerPeer)
     self.tracker.removeListener('update', self._onTrackerAnnounce)
     tasks.push(function (cb) {
-      self.tracker.destroy(cb)
+      process.nextTick(function () {
+        self.tracker.destroy(cb)
```

Comment:
The socket.close() method in cleanup() isn't enough to fully cleanup the udp tracker client. There is a setInterval and some other stuff that needs to cleaned up too, or else there will be memory leaks.

> I experimented last night and it works.

Yep, it works from the server's perspective (the STOP event is received), but not calling the destroy() method isn't gonna fly.

I have a large PR coming in a minute that fixes all the client implementations (not just udp) so they wait up to 1s for pending requests to be sent before force destroying everything.